tools/testrunner/testrunner.go

```diff
@@ -38,116 +37,27 @@ import (
 	"strings"
 
 	"github.com/google/uuid"
-	"github.com/jstemmer/go-junit-report/v2/junit"
 )
 
-type node struct {
-	children map[string]node
-}
-
-func (n node) visitor(path ...string) func(yield func(string, node) bool) {
-	return func(yield func(string, node) bool) {
-		for name, child := range n.children {
-			path := append(path, name)
-			if !yield(strings.Join(path, "/"), child) {
-				return
-			}
-			child.visitor(path...)(yield)
-		}
-	}
-}
-
-func (n node) walk() iter.Seq2[string, node] {
-	return n.visitor()
-}
-
 type attempt struct {
-	runner       *runner
-	number       int
-	junitXmlPath string
-	exitErr      *exec.ExitError
-	suites       junit.Testsuites
+	runner      *runner
+	number      int
+	exitErr     *exec.ExitError
+	junitReport *junitReport
 }
 
-func (a *attempt) run(ctx context.Context, args []string) error {
-	args = append([]string{"--junitfile", a.junitXmlPath}, args...)
+func (a *attempt) run(ctx context.Context, args []string) (error, string) {
```

Comment:
This is very non-standard to return the error first.

Reply:
Good point